File: test_api.js
// Simple test script for API endpoints
const API_BASE_URL = 'https://perfectcv-production.up.railway.app/api/v1';
const TOKEN = ''; // Add your token here

// Function to test an API endpoint
async function testEndpoint(endpoint, method = 'GET', data = null, headers = {}) {
  const url = `${API_BASE_URL}/${endpoint}`;
  const defaultHeaders = {
    'Authorization': `Bearer ${TOKEN}`,
    'Origin': 'https://perfect-cv-snowy.vercel.app',
    'Content-Type': 'application/json'
  };

  // Merge headers
  const finalHeaders = { ...defaultHeaders, ...headers };
  
  // Remove Content-Type if it's FormData
  if (data instanceof FormData) {
    delete finalHeaders['Content-Type'];
  }

  console.log(`Testing ${method} ${url}`);
  console.log('Headers:', finalHeaders);
  
  if (data instanceof FormData) {
    console.log('FormData contents:', Array.from(data.entries()));
  } else {
    console.log('Data:', data);
  }

  try {
    const options = {
      method,
      headers: finalHeaders,
      credentials: 'include'
    };

    if (data) {
      if (data instanceof FormData) {
        options.body = data;
      } else {
        options.body = JSON.stringify(data);
      }
    }

    const response = await fetch(url, options);
    console.log('Status:', response.status);
    console.log('Response Headers:', Object.fromEntries([...response.headers]));
    
    const contentType = response.headers.get('content-type');
    if (contentType && contentType.includes('application/json')) {
      const jsonData = await response.json();
      console.log('Response Body:', jsonData);
      return jsonData;
    } else {
      const textData = await response.text();
      console.log('Response Body (text):', textData);
      return textData;
    }
  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

// Test resume upload
async function testResumeUpload() {
  // Create a FormData object
  const formData = new FormData();
  
  // Add a file to the FormData
  // Note: In a browser environment, you would get this from a file input
  // For this test script, we'll simulate it with a text file
  const fileContent = 'This is a test resume content';
  const file = new Blob([fileContent], { type: 'application/pdf' });
  formData.append('file', file, 'test_resume.pdf');
  formData.append('title', 'Test Resume');
  
  // Test the upload endpoint
  return await testEndpoint('resumes/upload', 'POST', formData);
}

// Test optimization
async function testOptimization(resumeId) {
  return await testEndpoint('optimizations/optimize-resume', 'POST', {
    resume_id: resumeId,
    job_description_id: 0
  });
}

// Main function to run tests
async function runTests() {
  try {
    console.log('Testing Resume Upload API');
    console.log('=========================');
    
    // Test resume upload
    const uploadResult = await testResumeUpload();
    console.log('Upload Result:', uploadResult);
    
    // If upload was successful, test optimization
    if (uploadResult && uploadResult.id) {
      console.log('\nTesting Optimization API');
      console.log('=========================');
      
      const optimizationResult = await testOptimization(uploadResult.id);
      console.log('Optimization Result:', optimizationResult);
    }
  } catch (error) {
    console.error('Test failed:', error);
  }
}

// Run the tests
runTests(); 

File: test_api.py
import requests
import json
from datetime import datetime

def test_api():
    base_url = "http://localhost:8080/api/v1"
    
    # Test registration
    print("\nTesting registration...")
    register_data = {
        "email": f"test_{datetime.now().strftime('%Y%m%d%H%M%S')}@example.com",
        "password": "testpassword123",
        "full_name": "Test User"
    }
    
    try:
        response = requests.post(
            f"{base_url}/auth/register",
            json=register_data
        )
        print(f"Registration status: {response.status_code}")
        print(f"Response: {response.text}")
        
        if response.status_code == 200:
            print("Registration successful!")
            user_data = response.json()
            
            # Test login
            print("\nTesting login...")
            login_data = {
                "username": register_data["email"],
                "password": register_data["password"]
            }
            
            response = requests.post(
                f"{base_url}/auth/login",
                data=login_data
            )
            print(f"Login status: {response.status_code}")
            print(f"Response: {response.text}")
            
            if response.status_code == 200:
                print("Login successful!")
                token_data = response.json()
                
                # Test /me endpoint
                print("\nTesting /me endpoint...")
                headers = {"Authorization": f"Bearer {token_data['access_token']}"}
                response = requests.get(f"{base_url}/auth/me", headers=headers)
                print(f"Me endpoint status: {response.status_code}")
                print(f"Response: {response.text}")
                
                if response.status_code == 200:
                    print("Me endpoint test successful!")
                else:
                    print("Me endpoint test failed!")
            else:
                print("Login failed!")
        else:
            print("Registration failed!")
            
    except requests.exceptions.RequestException as e:
        print(f"Network error: {str(e)}")
    except Exception as e:
        print(f"Unexpected error: {str(e)}")

if __name__ == "__main__":
    test_api() 